source/use-cases/local-master-key.txt

```diff
@@ -1,23 +1,48 @@
 A. Create a Local Master Key
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Client-Side Field Level Encryption for MongoDB uses an encryption strategy called envelope encryption in which keys used to encrypt/decrypt data (called **data encryption keys**) are encrypted with another key (called the **master key**). 
+MongoDB Client-Side Field Level Encryption (CSFLE) uses an encryption strategy called envelope encryption in which keys used to encrypt/decrypt data (called **data encryption keys**) are encrypted with another key (called the **master key**). For more information on the features of envelope encryption and key management concepts, see `AWS Key Management Service Concepts <https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#enveloping`_.
 
-This strategy bolsters the security, minimizes encryption/decryption overhead, and allows the data encryption keys to be stored in the same replica set as the encrypted data. If a basic encryption strategy was used instead of envelope encryption, the computational and network load of encryption and decryption would need to be handled by the system that manages your keys. For more information on envelope encryption and key management concepts, read `this AWS article <https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#enveloping>`_.
+In CSFLE, the master key, used by the MongoDB driver to create and encrypt data keys, is securely stored in a Key Management Service (KMS). The data encryption keys, used to encrypt document fields, is stored in a user-specified collection alongside the encrypted data. 
```

Comment:
It's preferred to use a KMS but it doesn't necessarily use one - our first is all about local master keys.

e.g.
```rst
MongoDB drivers use the master key, which is stored either locally or in
a KMS, to create and encrypt data keys. They then store the encrypted
data keys in a user-specified collection in the same database as the
encrypted data.
```

Reply:
nit: data encryption keys ... _is_ stored ->... _are_ stored

Reply:
Sorry, that _are_ my bad. Do we have any licenses to Grammar.ly? After revising a few times, proofreading becomes tougher.